### DISCUSSION TITLE
What is the rate limit of this proxy server? How to replace it with your own proxy server?

### DISCUSSION THREAD

**Discussion by luluhuhu** [acheong08/ChatGPT-Proxy-V4#81] [Q&A] [answered]
What is the rate limit of this proxy server?
How to replace it with your own proxy server?

**Comment by acheong08** [comment, marked as answer]
Environment variable: `CHATGPT_BASE_URL`

`export CHATGPT_BASE_URL="https://yourserver.com/api/"`

> What is the rate limit of this proxy server?

With PUID, none IP based. There are account based rate limits

**Reply by luluhuhu** [reply]
Thank you for your answer.
My proxy server address is api.openai.com:4433,
This is my setup, but it reports an error:

echo $CHATGPT_BASE_URL
https://api.openai.com:4433/api/

curl https://api.openai.com:4433/api/conversation
{
  "error": {
    "message": "Invalid URL (GET /api/conversation)",
    "type": "invalid_request_error",
    "param": null,
    "code": null
  }
}

Calling revChatgpt also reported an error:
HTTPSConnectionPool(host='api.openai.com', port=4433): Max retries exceeded with url: /api/conversation (Caused by SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:1002)')))

**Reply by acheong08** [reply]
It is impossible for your API address to be api.openai.com...

**Reply by luluhuhu** [reply]
Sorry, I missed a piece of information.
I configured in /etc/hosts:
My server IP api.openai.com

ChatGPT-Proxy-V4: 
What is the speed limit based on IP?
How many requests per minute or per hour?

**Reply by acheong08** [reply]
50 requests per hour per free account.

**Reply by luluhuhu** [reply]
Excuse me,
I am not running ChatGPT-Proxy-V4 on my proxy server,
Do I need to download and run the code of ChatGPT-Proxy-V4 on my proxy server?

**Reply by luluhuhu** [reply]
I installed and run ChatGPT-Proxy-V4 on my server, the port is 9090,

curl http://server IP:9090/api/conversation:
Returned Sorry, you have been blocked,

Calling revChatgpt returns: OpenAI: (code: 502),
What is the reason?

**Reply by acheong08** [reply]
Cloudflare IP range block

**Reply by luluhuhu** [reply]
Thank you for your answer.
Is the solution to change the IP of the server?

**Reply by acheong08** [reply]
Yes. It seems DigitalOcean (Singapore server) is whitelisted

**Reply by luluhuhu** [reply]
I created a server in Singapore and deployed ChatGPT-Proxy-V4 on it, but still reported the same error, how to solve it?
